### DISCUSSION TITLE
Behavior of `onBeforePrerenderStart` hook

### DISCUSSION THREAD

**Discussion by marek-hanzal** [vikejs/vike#1695] [Discussion]
Hello,

I'm using Vike for SSG with slug in url `/@locale`. The problem is every nested route calls root `onBeforePrerenderStart` and fails as it returns the same locale.

One workaround is to put `onBeforePrerenderStart` in every nested directory, **which is not acceptable** for many reasons.

Am I using this hook wrong, is there some workaround, or is there some hope this hook get's it's URL, so it would be possible to properly generate all pages using one hook?

![image](https://github.com/vikejs/vike/assets/13855126/0649e699-6e26-4851-be58-983060e43b94)

Thank you!

P.S. I've found quite dirty hack able to somehow fix the problem:
(`runPrerenderer.js`)
- just provide pageConfig which already is there
![image](https://github.com/vikejs/vike/assets/13855126/dbcf8d02-0051-48a4-8c22-e1dc15368d3f)

Then I can do something like this:
![image](https://github.com/vikejs/vike/assets/13855126/6cdd1ab2-e8d3-4eb0-88dd-af2bc5963a5a)

There is probably nicer way how to solve this, but goal is just to start a .... discussion :).

**Comment by brillout** [comment]
I don’t see how your use case is different from the usual i18n use case so I guess following the i18n docs may help you, namely using both hooks onBeforePrerenderStart as well as onPrerenderStart.

**Reply by marek-hanzal** [reply]
Hmm... truth is I've directly ported code from `next.js` and I didn't look into i18n docs deeply, I'll check it out, thank you.
